From b101ff56e874824fa9f0d37a8468b07dbf3d002c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Tue, 16 Aug 2016 11:54:38 +0100
Subject: [PATCH] Resolves: tdf#101165 crash on deselecting all filters

this affects 5-1, but not master because of

7967e5e51e5210b8c3d3dc63502bd7d875eb36b7 which changed
the trigger in ScCheckListMenuWindow::setAllMemberState
but lets fix it in master anyway in case there's another
route in here

Change-Id: I49162bb73bf6dbef5cff68d35d10da2c47d9f2b5
---
 sc/source/ui/cctrl/checklistmenu.cxx | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sc/source/ui/cctrl/checklistmenu.cxx b/sc/source/ui/cctrl/checklistmenu.cxx
index 93f9aef..a017cda 100644
--- a/sc/source/ui/cctrl/checklistmenu.cxx
+++ b/sc/source/ui/cctrl/checklistmenu.cxx
@@ -1714,8 +1714,11 @@ SvTreeListEntry* ScCheckListBox::ShowCheckEntry( const OUString& sName, ScCheckL
     {
         if ( !pEntry )
         {
-            if ( rMember.mbDate )
+            if (rMember.mbDate)
             {
+                if (rMember.maDateParts.empty())
+                    return nullptr;
+
                 SvTreeListEntry* pYearEntry = FindEntry( nullptr, rMember.maDateParts[0] );
                 if ( !pYearEntry )
                     pYearEntry = InsertEntry( rMember.maDateParts[0], nullptr, true );
-- 
2.7.4

